File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> hashS;
        unordered_map<char,int> hashT;

        for(int i = 0;i<s.size();i++)hashS[s[i]]++;
        for(int i = 0;i<t.size();i++)hashT[t[i]]++;

        for(int i = 0;i<s.size();i++){
            if(hashS[s[i]] != hashT[s[i]])return false;
        }
        for(int i = 0;i<t.size();i++){
            if(hashS[t[i]] != hashT[t[i]])return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # First Approach 

        # newS = sorted(s)
        # newT = sorted(t)

        # if newS == newT :
        #     return True 

        # return False

        hashSet = defaultdict()

        if len(s) != len(t) :
            return False

        for charS , charT in zip(s,t) :
            hashSet[charS] = hashSet.get(charS , 0) + 1

            hashSet[charT] = hashSet.get(charT , 0) - 1

        if not any(hashSet.values()) : 
            return True 


        return False 
 